### PR TITLE
Updates for ncalc

### LINKED ISSUE #124: Connect PSALogin shared component to NCalc
**Is your feature request related to a problem? Please describe.**
we need to connect the shared component PSALogin to Ncalc

**Describe the solution you'd like**
when the NCalc application has implemented the shared PSALogin component

### LINKED ISSUE #126: Updates for ncalc
- [x] Update the dependencies, remove duplicate dependencies for shared components
- [x] replace the ncalc map with shared component 
- [x] Update api for feedback and models
- [x] Integrate shared header and create stepper
- [x] satellite api getting overload
- [x] look into the field select dropdown
- [ ] divide redux store to groups of data
- [ ] get familiar to the biomass api

### PR DESCRIPTION
**Pull request by YazidZidane**
- Update the dependencies, remove duplicate dependencies for shared components
- replace the ncalc map with shared component
- Update api for feedback and models
- Integrate shared header and create stepper
- resolved satellite api getting overload
- look into the field select dropdown
- Updated field drop down select function

### CODE REVIEW

**Comment by mikahpinegar** [comment]
i ran `npx unimported` and found 6 unimported files and one unimported module. can you verify these are not needed?

─────┬─────────────────
  │ 1 unused dependencies
─────┼─────────────────
   1 │ sass
─────┴─────────────────


─────┬─────────────────
   │ 6 unimported files
─────┼─────────────────
   1 │ src/components/Output/subcomponents/error.jsx
   2 │ src/components/Output/subcomponents/Loading.jsx
   3 │ src/components/ResponsiveNavBar/index.jsx
   4 │ src/components/SatPath/subcomponents/BiomassMapWidget.jsx
   5 │ src/components/SatPath/subcomponents/error.jsx
   6 │ src/components/SatPath/subcomponents/Loading.jsx

File: src/components/CashCrop/index.jsx [hunk 1]
```diff
@@ -154,17 +130,20 @@ const CashCrop = ({ barebone = false }) => {
               }}
               mt={6}
             >
-              <NavButton onClick={() => navigate('/covercrop')}>BACK</NavButton>
               <NavButton
-                onClick={() => navigate('/output')}
-                disabled={
```
Comment (mikahpinegar) [inline src/components/CashCrop/index.jsx:160]: why did this disabled logic get removed?

**Comment by YazidZidane** [comment]
> i ran `npx unimported` and found 6 unimported files and one unimported module. can you verify these are not needed?
> 
> ─────┬───────────────── │ 1 unused dependencies ─────┼───────────────── 1 │ sass ─────┴─────────────────
> 
> ─────┬───────────────── │ 6 unimported files ─────┼───────────────── 1 │ src/components/Output/subcomponents/error.jsx 2 │ src/components/Output/subcomponents/Loading.jsx 3 │ src/components/ResponsiveNavBar/index.jsx 4 │ src/components/SatPath/subcomponents/BiomassMapWidget.jsx 5 │ src/components/SatPath/subcomponents/error.jsx 6 │ src/components/SatPath/subcomponents/Loading.jsx

Yes, all the files under `SatPath` and `ResponsiveNavbar` are previous components and will not be used now. I'm just keeping them because I'm not sure if they'll be used in the future. The file under `Output` is not being used and can be deleted.

File: src/shared/Inputs/index.jsx [hunk 2]
```diff
@@ -1,6 +1,11 @@
-import React, { useEffect, useState, useRef, useCallback } from 'react';
+import React, {
+  useEffect, useState, useRef, useCallback,
+} from 'react';
 import { useSelector, useDispatch } from 'react-redux';
-import { TextField, Radio, RadioGroup, FormControlLabel, FormLabel, Autocomplete as MUIAutocomplete } from '@mui/material';
+import {
+  TextField, Radio, RadioGroup, FormControlLabel, FormLabel, Autocomplete as MUIAutocomplete,
```
Comment (mikahpinegar) [inline src/shared/Inputs/index.jsx:6]: should this use PSARadiobutton?

File: src/components/CashCrop/index.jsx (hunk 1 above)
Comment (YazidZidane) [reply src/components/CashCrop/index.jsx:160]: They are not removed, it's still in the Next button's onClick. This line is aligned to the Back button

File: src/shared/Inputs/index.jsx (hunk 2 above)
Comment (YazidZidane) [reply src/shared/Inputs/index.jsx:6]: This component is replaced in the user history pr and will not be used anymore

File: src/components/Output/subcomponents/NitrogenWidget.jsx [hunk 3]
```diff
@@ -137,8 +139,18 @@ const NitrogenCard = ({ refVal }) => {
   useEffect(() => {
     if (chartRef1.current && chartRef2.current) {
       if (!surfaceData || surfaceData.length === 0) {
-        chartRef1.current.chart.showLoading();
-        chartRef2.current.chart.showLoading();
+        const spinnerHtml = ReactDOMServer.renderToString(
```
Comment (mikahpinegar) [inline src/components/Output/subcomponents/NitrogenWidget.jsx:142]: why is this needed on the spinner component?

File: src/components/ResponsiveNavBar/index.jsx [hunk 4]
```diff
@@ -105,10 +107,9 @@ const ResponsiveNavBar = ({ screens }) => {
 
   // // useSelector(get.screen); // force render
   useEffect(() => {
-    setActiveMenu(
-      location.pathname.replace('/', '').replace('2', ''),
-    );
+    setActiveMenu(location.pathname.replace('/', '').replace('2', ''));
   }, [location]);
+  console.log('123', screens);
```
Comment (mikahpinegar) [inline src/components/ResponsiveNavBar/index.jsx]: remove clg

File: src/shared/Map/BiomassMap.jsx [hunk 5]
```diff
@@ -31,6 +31,8 @@ const BiomassMapComp = ({ variant }) => {
   const [features, setFeatures] = useState(mapPolygon);
   const [drawEvent, setDrawEvent] = useState({});
 
+  const mapboxToken = import.meta.env.VITE_MAPBOX_TOKEN;
```
Comment (mikahpinegar) [inline src/shared/Map/BiomassMap.jsx:34]: can you make a keys.js file similar to the selector and then import the env files from there to be consisten with other tools?

File: src/shared/Map/NitrogenMap.jsx [hunk 6]
```diff
@@ -31,7 +32,10 @@ const NitrogenMapComp = ({ variant }) => {
   const [features, setFeatures] = useState(mapPolygon);
   const [drawEvent, setDrawEvent] = useState({});
 
+  const mapboxToken = import.meta.env.VITE_MAPBOX_TOKEN;
```
Comment (mikahpinegar) [inline src/shared/Map/NitrogenMap.jsx:35]: can you make a keys.js file similar to the selector and then import the env files from there to be consisten with other tools?

File: src/components/Output/subcomponents/NitrogenWidget.jsx (hunk 3 above)
Comment (YazidZidane) [reply src/components/Output/subcomponents/NitrogenWidget.jsx:142]: @ravighevariya10

File: src/shared/Map/BiomassMap.jsx (hunk 5 above)
Comment (YazidZidane) [reply src/shared/Map/BiomassMap.jsx:34]: Yes, such changes have been made in the user history pr

File: src/components/Output/subcomponents/NitrogenWidget.jsx (hunk 3 above)
Comment (ravighevariya10) [reply src/components/Output/subcomponents/NitrogenWidget.jsx:142]: It's needed because showLoading() expects an HTML string to be passed and to integrate our shared loading-indicator, we have to pass it as string HTML so this 'ReactDOMServer.renderToString()' function converts the React component to an HTML string
